lib/ex_auction/auction/worker.ex

```diff
@@ -42,19 +42,19 @@ defmodule ExAuction.Auction.Worker do
     end
   end
 
-  def start_link(%ExAuction.Auction{} = auction) do
+  def start_link(%__MODULE__.State{auction: %ExAuction.Auction{} = auction} = auction_state) do
     case whereis(auction) do
       nil ->
-        {:ok, pid} = GenServer.start_link(__MODULE__, auction)
+        {:ok, pid} = GenServer.start_link(__MODULE__, auction_state)
         register_process(pid, auction)
 
       pid ->
         {:error, {:already_started, pid}}
     end
   end
 
-  def init(auction) do
-    state = %__MODULE__.State{auction: %Auction{auction | status: :active}, bids: []}
+  def init(%__MODULE__.State{auction: auction, bids: init_state_bids} = _auction_state) do
+    state = %__MODULE__.State{auction: %Auction{auction | status: :active}, bids: init_state_bids}
```

Comment:
```
%__MODULE__.State{auction: %Auction{auction | status: :active}, bids: init_state_bids} = state
```

Reply:
What is the idea here?

Are you suggesting that I match `state` by flipping `LH= RH`?